scriptmodules/supplementary/sixaxis.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="sixaxis"
+rp_module_desc="Helper service to install & configure the latest drivers for official and third-party DualShock controllers (ps3controller replacement)\n\nNote: For Shanwan/GASIA third-party controllers, enable third-party support in the configuration options.\n\nTo pair controllers, use the RetroPie Bluetooth menu, choose 'Register and Connect...', then follow the on-screen instructions."
+rp_module_licence="GPL2 https://raw.githubusercontent.com/RetroPie/sixaxis/master/COPYING"
+rp_module_section="driver"
+
+function depends_sixaxis() {
+    local depends=(checkinstall libevdev-tools)
+
+    getDepends "${depends[@]}"
+
+    rp_callModule ps3controller remove
+}
+
+function sources_sixaxis() {
+    gitPullOrClone "$md_build/sixaxis" https://github.com/RetroPie/sixaxis.git
+}
+
+function build_sixaxis() {
+    cd sixaxis
+    make clean
+    make
+    md_ret_require+=("$md_build/sixaxis/bins/sixaxis-timeout")
+}
+
+function gui_sixaxis() {
+    local sixaxis_config="$md_conf_root/all/sixaxis_timeout.cfg"
+    local cmd=(dialog --backtitle "$__backtitle" --menu "Choose an option." 22 86 16)
+    local options=(
+        1 "Enable support for third-party controllers"
+        2 "Disable support for third-party controllers"
+        3 "Configure controller timeout"
+    )
+    local choice=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+    if [[ -n "$choice" ]]; then
+        case "$choice" in
+            1)
+                rp_callModule "customhidsony"
+                ;;
+            2)  rp_callModule "customhidsony" remove
+                rp_callModule "custombluez" remove
+                ;;
+            3)
+                if [[ -f "$sixaxis_config" ]]; then
+                    editFile "$sixaxis_config"
+                fi
+                ;;
+        esac
+    fi
+}
+
+function install_sixaxis() {
+    cd sixaxis
+    checkinstall -y --fstrans=no
+}
+
+function configure_sixaxis() {
+    local sixaxis_config="$md_conf_root/all/sixaxis_timeout.cfg"
+
+    if [[ "$md_mode" == "install" ]] && [[ ! -f "$sixaxis_config" ]]; then
+        cat > "$sixaxis_config" << _EOF_
+# Set your preferred controller timeout in seconds (0 to disable)
+SIXAXIS_TIMEOUT=600
+_EOF_
```

Comment:
You could use the copyDefaultConfig if you wanted. Create the config file in tmp via mktemp and then use that function. Then if they changed the settings they get a rp-dist file they can reference. 

Not an issue either way.